src/utils/FileEmitter.ts

```diff
@@ -241,24 +242,20 @@ export class FileEmitter {
 				reserveFileNameInBundle(emittedFile.fileName, this.bundle, this.options.onwarn);
 			}
 		}
-		for (const [referenceId, consumedFile] of this.filesByReferenceId.entries()) {
+		for (const [referenceId, consumedFile] of this.filesByReferenceId) {
 			if (consumedFile.type === 'asset' && consumedFile.source !== undefined) {
 				this.finalizeAsset(consumedFile, consumedFile.source, referenceId, this.bundle);
 			}
 		}
 	};
 
 	private assignReferenceId(file: ConsumedFile, idBase: string): string {
-		let referenceId: string | undefined;
+		let referenceId = idBase;
+
 		do {
-			const hash = createHash();
-			if (referenceId) {
-				hash.update(referenceId);
-			} else {
-				hash.update(idBase);
-			}
-			referenceId = hash.digest('hex').substr(0, 8);
+			referenceId = createHash().update(referenceId).digest('hex').substring(0, 8);
```

Comment:
```suggestion
			referenceId = createHash().update(referenceId || idBase).digest('hex').substring(0, 8);
```

Reply:
nice, thank you! missed the loop 😞

Reply:
I also removed the initial `idBase` assignment and used `??` as well.